pkg/xds/generator/secrets_generator.go

```diff
@@ -14,121 +16,106 @@ const OriginSecrets = "secrets"
 type SecretsProxyGenerator struct {
 }
 
-var _ ResourceGenerator = SecretsProxyGenerator{}
+var _ generator_core.ResourceGenerator = SecretsProxyGenerator{}
 
-type MeshCa struct {
-	Mesh     string
-	CaSecret *core_xds.CaSecret
+func CreateCaSecretResource(name string, ca *core_xds.CaSecret) *core_xds.Resource {
+	caSecret := envoy_secrets.CreateCaSecret(ca, name)
+	return &core_xds.Resource{
+		Name:     caSecret.Name,
+		Origin:   OriginSecrets,
+		Resource: caSecret,
+	}
 }
 
-func CreateSecretResources(
-	mesh *core_mesh.MeshResource,
-	identity *core_xds.IdentitySecret,
-	cas []xds_secrets.MeshCa,
-) *core_xds.ResourceSet {
-	resources := core_xds.NewResourceSet()
-
-	meshName := mesh.GetMeta().GetName()
-	identitySecret := envoy_secrets.CreateIdentitySecret(identity, meshName)
-
-	var caResources []*core_xds.Resource
-	for _, ca := range cas {
-		caSecret := envoy_secrets.CreateCaSecret(ca.CaSecret, ca.Mesh)
-		caResources = append(
-			caResources,
-			&core_xds.Resource{
-				Name:     caSecret.Name,
-				Origin:   OriginSecrets,
-				Resource: caSecret,
-			},
-		)
+func CreateIdentitySecretResource(name string, identity *core_xds.IdentitySecret) *core_xds.Resource {
+	identitySecret := envoy_secrets.CreateIdentitySecret(identity, name)
+	return &core_xds.Resource{
+		Name:     identitySecret.Name,
+		Origin:   OriginSecrets,
+		Resource: identitySecret,
 	}
-
-	resources.Add(
-		&core_xds.Resource{
-			Name:     identitySecret.Name,
-			Origin:   OriginSecrets,
-			Resource: identitySecret,
-		},
-	)
-	resources.Add(
-		caResources...,
-	)
-
-	return resources
 }
 
-func createDataPlaneProxySecrets(
+// GenerateSecretsFromTracker takes uses the SecretsTracker on Proxy and
+// generates whatever secrets were used in the config generation.
+func (g SecretsProxyGenerator) Generate(
 	ctx xds_context.Context,
 	proxy *core_xds.Proxy,
 ) (*core_xds.ResourceSet, error) {
-	mesh := ctx.Mesh.Resource
-
-	if !mesh.MTLSEnabled() {
-		return nil, nil
-	}
+	resources := core_xds.NewResourceSet()
 
-	var otherMeshes []*core_mesh.MeshResource
-	for _, otherMesh := range ctx.Mesh.Resources.CrossMeshGateways(mesh) {
-		if otherMesh.Mesh.GetMeta().GetName() != mesh.GetMeta().GetName() {
-			otherMeshes = append(otherMeshes, otherMesh.Mesh)
-		}
+	log := core.Log.WithName("secrets-generator")
+	switch {
+	case proxy.Dataplane != nil:
+		log = log.WithValues("proxy", proxy.Dataplane.GetMeta().GetName())
```

Comment:
mesh is fairly important here. How about instead of all cases etc. just adding "proxyID". It's pretty obvious is this is ingress or egress or regular data plane proxy.

Reply:
added `"type"`